src/course-unit/data/utils.js

```diff
@@ -0,0 +1,191 @@
+  return {
+    sequence: {
+      id: sequence.item_id,
+      blockType: sequence.tag,
+      unitIds: sequence.items.map(unit => unit.id),
+      bannerText: sequence.banner_text,
+      format: sequence.format,
+      title: sequence.display_name,
+      /*
+            Example structure of gated_content when prerequisites exist:
+            {
+              prereq_id: 'id of the prereq section',
+              prereq_url: 'unused by this frontend',
+              prereq_section_name: 'Name of the prerequisite section',
+              gated: true,
+              gated_section_name: 'Name of this gated section',
+            */
+      gatedContent: camelCaseObject(sequence.gated_content),
+      isTimeLimited: sequence.is_time_limited,
+      isProctored: sequence.is_proctored,
+      isHiddenAfterDue: sequence.is_hidden_after_due,
+      // Position comes back from the server 1-indexed. Adjust here.
+      activeUnitIndex: sequence.position ? sequence.position - 1 : 0,
+      saveUnitPosition: sequence.save_position,
+      showCompletion: sequence.show_completion,
+      allowProctoringOptOut: sequence.allow_proctoring_opt_out,
+    },
+    units: sequence.items.map(unit => ({
+      id: unit.id,
+      sequenceId: sequence.item_id,
+      bookmarked: unit.bookmarked,
+      complete: unit.complete,
+      title: unit.page_title,
+      contentType: unit.type,
+      graded: unit.graded,
+      containsContentTypeGatedContent: unit.contains_content_type_gated_content,
+    })),
+  };
+}
+
+export function normalizeLearningSequencesData(learningSequencesData) {
+  const models = {
+    courses: {},
+    sections: {},
+    sequences: {},
+  };
+
+  const now = new Date();
+  function isReleased(block) {
+    // We check whether the backend marks this as accessible because staff users are granted access anyway.
+    // Note that sections don't have the `accessible` field and will just be checking `effective_start`.
+    return block.accessible || !block.effective_start || now >= Date.parse(block.effective_start);
+  }
+
+  // Sequences
+  Object.entries(learningSequencesData.outline.sequences).forEach(([seqId, sequence]) => {
+    if (!isReleased(sequence)) {
+      return; // Don't let the learner see unreleased sequences
+    }
+
+    models.sequences[seqId] = {
+      id: seqId,
+      title: sequence.title,
+    };
+  });
+
+  // Sections
+  learningSequencesData.outline.sections.forEach(section => {
+    // Filter out any ignored sequences (e.g. unreleased sequences)
+    const availableSequenceIds = section.sequence_ids.filter(seqId => seqId in models.sequences);
+
+    // If we are unreleased and already stripped out all our children, just don't show us at all.
+    // (We check both release date and children because children will exist for an unreleased section even for staff,
+    // so we still want to show this section.)
+    if (!isReleased(section) && availableSequenceIds.length === 0) {
```

Comment:
[optional] We can simplify this condition a bit
```suggestion
    if (!isReleased(section) && !availableSequenceIds.length) {
```